### PR TITLE
Update Developer guide with mark and unmark implementation

### LINKED ISSUE #146: Update Developer Guide with design details
- [x] #168 
- [x] #160
- [x] #151
- [x] #153

### LINKED ISSUE #160: Update DG with Mark and Unmark implementation

### PR DESCRIPTION
**Pull request by Angelyxx**
closes #160

### CODE REVIEW

File: docs/images/ModelClassDiagram.png
Comment (geoffong11) [inline docs/images/ModelClassDiagram.png:1]: the `*filtered` seems to be at the wrong arrow (pointing to Person)

**Review by geoffong11 — COMMENTED** [review]
For ModelClassDiagram, the `*filtered` looks a bit out of place.

File: docs/images/ModelClassDiagram.png
Comment (Angelyxx) [reply docs/images/ModelClassDiagram.png:1]: updated this by adding a hidden arrow

**Review by geoffong11 — APPROVED** [review]
LGTM!

**Pull request merged**